cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1433,6 +1545,13 @@ data TxValidityLowerBound era where
 deriving instance Eq   (TxValidityLowerBound era)
 deriving instance Show (TxValidityLowerBound era)
 
+instance Semigroup (TxValidityLowerBound era) where
+    TxValidityNoLowerBound <> x = x
+    x <> TxValidityNoLowerBound = x
+    (TxValidityLowerBound wit l) <> (TxValidityLowerBound _ r) = TxValidityLowerBound wit (max l r)
```

Comment:
The thing that bothers me is we through away the witness on the RHS of `<>`.  Is this safe to do?

Reply:
Oh it's a type witness.  Maybe it's okay?

Reply:
It is safe to do for all the "feature supported in era" type witnesses. For any given era, they have at most one non-bottom value, so it doesn't matter "which one" you pick.